components/login/provider/index.jsx

```diff
@@ -24,30 +24,40 @@
 import React, { createRef, useEffect, useState } from "react";
 import T from "prop-types";
 import {
-  Box,
   FormControl,
   FormHelperText,
   TextField,
   useTheme,
 } from "@material-ui/core";
+import { makeStyles, createStyles } from "@material-ui/styles";
 import { Autocomplete } from "@material-ui/lab";
-
+import { useBem } from "@solid/lit-prism-patterns";
 import { LoginButton, useSession } from "@inrupt/solid-ui-react";
 
 import { Button } from "@inrupt/prism-react-components";
 import { generateRedirectUrl } from "../../../src/windowHelpers";
 import getIdentityProviders from "../../../constants/provider";
 import { ERROR_REGEXES, hasError } from "../../../src/error";
 import useIdpFromQuery from "../../../src/hooks/useIdpFromQuery";
+import styles from "./styles";
+
+const useStyles = makeStyles((theme) => createStyles(styles(theme)));
 
 const providers = getIdentityProviders();
-const TESTCAFE_ID_LOGIN_TITLE = "login-title";
 export const TESTCAFE_ID_LOGIN_FIELD = "login-field";
-const TESTCAFE_ID_LOGIN_BUTTON = "login-button";
+const TESTCAFE_ID_GO_BUTTON = "go-button";
 
 export function setupOnProviderChange(setProviderIri) {
   return (e, newValue) => {
-    setProviderIri(newValue);
+    if (typeof newValue === "string") {
+      if (newValue.startsWith("https://")) {
```

Comment:
This gets weird if a user plugs in `http://` instead of `https://`; the textbox doesn't reflect the value being used. For example, `http://inrupt.com` becomes `https://http://inrupt.com`.